Index: components/admin/HomeAdmin.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import React from 'react';\nimport Introduction from \"@/components/admin/Introduction\";\n\ninterface User {\n    id: string;\n    email: string;\n    name: string;\n    nBEvents: number;\n    isAdmin: boolean;\n    createdAt: string;\n}\ninterface HomeAdminProps {\n    userData: User;\n}\n\nconst HomeAdmin: React.FC<HomeAdminProps> = ({userData}) => {\n    return (\n        <>\n            <h1 className=\"animate-in text-3xl font-bold pb-5\">Bienvenue, {userData.name}</h1>\n            <Introduction/>\n        </>\n\n    );\n}\n\nexport default HomeAdmin;\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/components/admin/HomeAdmin.tsx b/components/admin/HomeAdmin.tsx
--- a/components/admin/HomeAdmin.tsx	(revision 00099c0ee0e4046078a5a0029eb783ee7b19accf)
+++ b/components/admin/HomeAdmin.tsx	(date 1717254991193)
@@ -10,7 +10,7 @@
     createdAt: string;
 }
 interface HomeAdminProps {
-    userData: User;
+    userData: any;
 }
 
 const HomeAdmin: React.FC<HomeAdminProps> = ({userData}) => {
Index: app/protected/page.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import { createClient } from \"@/utils/supabase/server\";\nimport { redirect } from \"next/navigation\";\nimport HomeAdmin from \"@/components/admin/HomeAdmin\";\nimport HomeCustomer from \"@/components/admin/HomeCustomer\";\n\nexport default async function ProtectedPage() {\n  const supabase = createClient();\n\n  const {\n    data: { user },\n  } = await supabase.auth.getUser();\n\n  if (!user) {\n    return redirect(\"/login\");\n  }\n  const {data:UserLogged} = await supabase.from('Users').select('*').eq('id', user.id).single()\n  const isAdmin = UserLogged.isAdmin\n\n  return (<div>\n        {isAdmin ? <HomeAdmin userData={UserLogged} /> : <HomeCustomer/>}\n      </div>\n\n  );\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/protected/page.tsx b/app/protected/page.tsx
--- a/app/protected/page.tsx	(revision 00099c0ee0e4046078a5a0029eb783ee7b19accf)
+++ b/app/protected/page.tsx	(date 1717255618057)
@@ -1,24 +1,45 @@
-import { createClient } from "@/utils/supabase/server";
-import { redirect } from "next/navigation";
-import HomeAdmin from "@/components/admin/HomeAdmin";
-import HomeCustomer from "@/components/admin/HomeCustomer";
+import HomeAdmin from '@/components/admin/HomeAdmin';
+import HomeCustomer from '@/components/admin/HomeCustomer';
+
+interface User {
+  id: string;
+  email: string;
+  name: string;
+  nBEvents: number;
+  isAdmin: boolean;
+  createdAt: string;
+}
+
+const ProtectedPage = () => {
 
-export default async function ProtectedPage() {
-  const supabase = createClient();
 
-  const {
-    data: { user },
-  } = await supabase.auth.getUser();
 
-  if (!user) {
-    return redirect("/login");
-  }
-  const {data:UserLogged} = await supabase.from('Users').select('*').eq('id', user.id).single()
-  const isAdmin = UserLogged.isAdmin
+
+  // useEffect(() => {
+  //   const fetchUser = async () => {
+  //     console.log('here')
+  //     const response = await fetch('/api/auth/getUser');
+  //     if (response.status === 401) {
+  //       router.push('/login');
+  //       return;
+  //     }
+  //
+  //     const data = await response.json();
+  //     setUser(data);
+  //     setLoading(false);
+  //   };
+  //
+  //   fetchUser();
+  // }, [router]);
+
+
 
-  return (<div>
-        {isAdmin ? <HomeAdmin userData={UserLogged} /> : <HomeCustomer/>}
+  return (
+      <div>
+          <h1>Protected Page</h1>
+            <HomeAdmin userData={{name: 'TestName'}} />
       </div>
-
   );
-}
+};
+
+export default ProtectedPage;
Index: app/api/auth/getUser.ts
===================================================================
diff --git a/app/api/auth/getUser.ts b/app/api/auth/getUser.ts
deleted file mode 100644
--- a/app/api/auth/getUser.ts	(revision 00099c0ee0e4046078a5a0029eb783ee7b19accf)
+++ /dev/null	(revision 00099c0ee0e4046078a5a0029eb783ee7b19accf)
@@ -1,24 +0,0 @@
-import { NextApiRequest, NextApiResponse } from 'next';
-import { createClient } from '@/utils/supabase/server';
-
-const supabase = createClient();
-
-export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-    const { data: { user }, error } = await supabase.auth.getUser();
-
-    if (error) {
-        return res.status(401).json({ error: 'Unauthorized' });
-    }
-
-    const { data: userLogged, error: userError } = await supabase
-        .from('Users')
-        .select('*')
-        .eq('id', user?.id)
-        .single();
-
-    if (userError) {
-        return res.status(401).json({ error: 'User not found' });
-    }
-
-    res.status(200).json(userLogged);
-}
diff --git a/components/admin/forms/UserInfosForm.tsx b/components/admin/forms/UserInfosForm.tsx
new file mode 100644
