Index: components/admin/HomeAdmin.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import React from 'react';\nimport Introduction from \"@/components/admin/Introduction\";\n\ninterface User {\n    id: string;\n    email: string;\n    name: string;\n    nBEvents: number;\n    isAdmin: boolean;\n    createdAt: string;\n}\ninterface HomeAdminProps {\n    userData: User;\n}\n\nconst HomeAdmin: React.FC<HomeAdminProps> = ({userData}) => {\n    return (\n        <>\n            <h1 className=\"animate-in text-3xl font-bold pb-5\">Bienvenue, {userData.name}</h1>\n            <Introduction/>\n        </>\n\n    );\n}\n\nexport default HomeAdmin;\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/components/admin/HomeAdmin.tsx b/components/admin/HomeAdmin.tsx
--- a/components/admin/HomeAdmin.tsx	(revision 349261f103cdf46f756daa2e6b0b0d1f6e29fca8)
+++ b/components/admin/HomeAdmin.tsx	(date 1717254800662)
@@ -10,7 +10,7 @@
     createdAt: string;
 }
 interface HomeAdminProps {
-    userData: User;
+    userData: any;
 }
 
 const HomeAdmin: React.FC<HomeAdminProps> = ({userData}) => {
Index: app/protected/page.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import { createClient } from \"@/utils/supabase/server\";\nimport { redirect } from \"next/navigation\";\nimport HomeAdmin from \"@/components/admin/HomeAdmin\";\nimport HomeCustomer from \"@/components/admin/HomeCustomer\";\nimport Header from \"@/components/admin/Header\";\n\nexport default async function ProtectedPage() {\n  const supabase = createClient();\n\n  const {\n    data: { user },\n  } = await supabase.auth.getUser();\n\n  if (!user) {\n    return redirect(\"/login\");\n  }\n  const {data:UserLogged} = await supabase.from('Users').select('*').eq('id', user.id).single()\n  const isAdmin = UserLogged.isAdmin\n\n  return (<div>\n        {isAdmin ? <HomeAdmin userData={UserLogged} /> : <HomeCustomer/>}\n      </div>\n\n  );\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/protected/page.tsx b/app/protected/page.tsx
--- a/app/protected/page.tsx	(revision 349261f103cdf46f756daa2e6b0b0d1f6e29fca8)
+++ b/app/protected/page.tsx	(date 1717254800692)
@@ -1,24 +1,36 @@
+'use client'
+
 import { createClient } from "@/utils/supabase/server";
 import { redirect } from "next/navigation";
 import HomeAdmin from "@/components/admin/HomeAdmin";
 import HomeCustomer from "@/components/admin/HomeCustomer";
 import Header from "@/components/admin/Header";
 
-export default async function ProtectedPage() {
+
+
+
+export default  function ProtectedPage() {
   const supabase = createClient();
 
-  const {
-    data: { user },
-  } = await supabase.auth.getUser();
 
-  if (!user) {
-    return redirect("/login");
+  async function getUser() {
+
+    const userLogged = await supabase.auth.getUser();
+    if (!user) {
+      return redirect("/login");
+    } else {
+      console.log('OK TROP BIEN', userLogged)
+    }
   }
-  const {data:UserLogged} = await supabase.from('Users').select('*').eq('id', user.id).single()
-  const isAdmin = UserLogged.isAdmin
+
+
+
+  // const {data:UserLogged} = await supabase.from('Users').select('*').eq('id', user.id).single()
+  // const isAdmin = UserLogged.isAdmin
+  const isAdmin = true
 
   return (<div>
-        {isAdmin ? <HomeAdmin userData={UserLogged} /> : <HomeCustomer/>}
+        {isAdmin ? <HomeAdmin userData={user} /> : <HomeCustomer/>}
       </div>
 
   );
